### PR TITLE
osd: PeeringState: fix selection order in calc_replicated_acting_stretch

### PR DESCRIPTION
**Pull request by gregsfortytwo**
We were previously mis-ordering these to *de*prioritize the existing acting set. That is bad!

We generate OSD candidates from the acting set and strays, and push
them into the candidates list as a tuple of <<!in_acting,pg_info.last_update>,osd_id>.

Then we sort the list. Then we go through the list from front to back and
push_back entries into the appropriate ancestor lists.

And then we pop_back() off the lists to select the acting set.
Which of course turns our nice careful order backwards! So don't do that.

Fixes: https://tracker.ceph.com/issues/53824

Signed-off-by: Greg Farnum <gfarnum@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Our lack of upstream tests around stretch mode is a problem here: I don't really see how to properly test this except giving it to the folks who have seen it (Red Hat QE; thanks guys!). So I think we just do our due diligence that it doesn't break the suite and then merge it in.

File: src/osd/PeeringState.cc [hunk 1]
```diff
@@ -1855,7 +1855,7 @@ class bucket_candidates_t {
   osd_id_t pop_osd() {
     ceph_assert(!is_empty());
     auto ret = osds.back();
```
Comment (athanatos) [inline src/osd/PeeringState.cc:1857]: Seems like this also needs to change?

Comment (gregsfortytwo) [reply src/osd/PeeringState.cc]: Whoops, I found this bug before Christmas and when I picked it back up forgot how much I'd done. Thanks!

**Review by athanatos — APPROVED** [review]

**Comment by gregsfortytwo** [comment]
No patch changes, just rebased to latest master since the doc and make check both failed on not-my-code?

**Review by neha-ojha — APPROVED** [review]

**Comment by neha-ojha** [comment]
jenkins test api

**Comment by neha-ojha** [comment]
> Our lack of upstream tests around stretch mode is a problem here: I don't really see how to properly test this except giving it to the folks who have seen it (Red Hat QE; thanks guys!). So I think we just do our due diligence that it doesn't break the suite and then merge it in.

what kind of test scenario uncovered this bug?

**Comment by yuriw** [comment]
jenkins test api

**Comment by yuriw** [comment]
jenkins test api

**Comment by neha-ojha** [comment]
jenkins test api

**Comment by ljflores** [comment]
The failed api test showed this line: `Building remotely on 172.21.2.12+braggi12` indicating that it was running on braggi12. There was an issue with braggi12 in the past, but it was fixed by David Galloway so not entirely sure it was the same thing.

In any case, we'll see if it passes now.

**Pull request merged**